src/main/kotlin/org/rust/lang/doc/psi/RsDocKind.kt

```diff
@@ -64,28 +65,40 @@ enum class RsDocKind {
      * This method expects non-empty line sequences of valid doc comment tokens.
      * It does **not** perform any validation!
      */
-    abstract fun removeDecoration(lines: Sequence<String>): Sequence<String>
+    protected abstract fun removeDecoration(lines: Sequence<RsDocLine>): Sequence<RsDocLine>
 
-    protected fun removeEolDecoration(decoratedLines: Sequence<String>, infix: String = this.infix): Sequence<String> {
-        val lines = decoratedLines.map { it.substringAfter(infix) }
-        val firstLineIndented = lines.first()
+    fun removeDecorationToLines(text: CharSequence): Sequence<RsDocLine> =
+        removeDecoration(RsDocLine.splitLines(text))
+
+    fun removeDecoration(text: CharSequence): Sequence<CharSequence> =
+        removeDecorationToLines(text).map { it.content }
+
+    protected fun removeEolDecoration(decoratedLines: Sequence<RsDocLine>, infix: String = this.infix): Sequence<RsDocLine> {
```

Comment:
I didn't change anything in this escaping logic (hence there is `RsDocLine.dropWhileAtMost` and others), but it's obviously buggy and should be rewritten later